src/openms/source/ANALYSIS/OPENSWATH/IonMobilityScoring.cpp

```diff
@@ -200,6 +210,141 @@ namespace OpenMS
     }
   }
 
+  OpenMS::Mobilogram sumAlignedMobilograms(const std::vector<OpenMS::Mobilogram>& aligned_mobilograms) 
+  {
+    if (aligned_mobilograms.empty()) {
+      return OpenMS::Mobilogram();
+    }
+
+    OpenMS::Mobilogram summed_mobilogram;
+
+    // Use the first mobilogram to set the structure
+    const auto& first_mobilogram = aligned_mobilograms[0];
+
+    for (size_t j = 0; j < first_mobilogram.size(); ++j) {
+      OpenMS::MobilityPeak1D summed_peak;
+      summed_peak.setMobility(first_mobilogram[j].getMobility());
+      summed_peak.setIntensity(0.0); 
+
+      // Sum intensities from all mobilograms
+      for (const auto& mobilogram : aligned_mobilograms) {
+        if (j < mobilogram.size()) {
+          summed_peak.setIntensity(summed_peak.getIntensity() + mobilogram[j].getIntensity());
+        }
+      }
+
+      summed_mobilogram.push_back(summed_peak);
+    }
+
+    // Concatenate names of all mobilograms
+    std::string concatenated_name;
+    for (size_t i = 0; i < aligned_mobilograms.size(); ++i) {
+      if (i > 0) {
+        concatenated_name += "|";
+      }
+      concatenated_name += aligned_mobilograms[i].getName();
+    }
+
+    // Set the concatenated name for the summed mobilogram
+    summed_mobilogram.setName(concatenated_name);
+
+    return summed_mobilogram;
+  }
+
+  std::tuple<size_t, size_t, size_t> findHighestPeak(const PeakPickerMobilogram& picker) 
+  {
+    const auto& intensities = picker.integrated_intensities_;
+
+    // Check if vectors are empty or of different sizes
+    if (intensities.empty() ||
+        intensities.size() != picker.left_width_.size() ||
+        intensities.size() != picker.right_width_.size()) {
+      // Return an "invalid" tuple if there's an issue
+      return std::make_tuple(std::numeric_limits<size_t>::max(), 0, 0);
```

Comment:
Yes, this sounds like a better solution.
What I would also probably do is trying to reduce the "requirements" of methods to the minimum.
E.g. here findHighestPeak(const PeakPickerMobilogram& picker) requires a full PeakPickerMobilogram though only the integrated intensities are used.
two better options:
1. either only pass the integrated intensities
2. or pass "PeakPickerMobilogram" and use some extra information (e.g. it could store if picking was successful)

of these two options number 1 is better for various reasons (e.g., doesn't rely on additional state (=easier to parallelize), is a simpler "interface" than the full PeakPickerMobilogram)